### ISSUE TITLE
TikTok videos with audio removed do not download

### ISSUE THREAD

**Issue by julian45** [yt-dlp/yt-dlp#4992]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.09.01** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States

### Provide a description that is worded well enough to be understood

Following up on a topic discussed in issue #3776, finishing where the conversation left off:

> > @pukkandan: would an extractor that's dependent on an external endpoint (either ran by the user or hosted elsewhere) be viable for yt-dlp? I wouldn't think so, based on a look at the other dependencies for this project. Evil0ctal's project basically just sets up an API endpoint that you can pass URLs to and get a video as the response.
> 
> No, but you can write a plugin for it and distribute that yourself

I figured, but I thought it might be a good idea to just have that question and answer on the record.

Besides, #4984 seems to cover most of the same ground that Evil0ctal's API would (downloading individual videos, incl. 720p, and with possibility of no watermark). The only thing it can do that bashonly's fix can't is imawin954's edge case of a video with audio removed by TikTok; as described, the page 404s when browsed to, but the video is accessible from the poster's user page

The command output below is from a binary of yt-dlp built at commit `163281178a61565cd592426d452978ff47e63439`, which is after #4984 was merged, using the same example video as imawin954.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['https://www.tiktok.com/@espnatbat/video/7096644298396978475', '-vU']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.09.01 [5d7c7d656] (zip)
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python 3.10.6 (CPython 64bit) - macOS-12.6-arm64-arm-64bit
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.1.1 (setts), ffprobe 5.1.1, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.14, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1672 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.09.01, Current version: 2022.09.01
yt-dlp is up to date (2022.09.01)
[debug] [TikTok] Extracting URL: https://www.tiktok.com/@espnatbat/video/7096644298396978475
[TikTok] 7096644298396978475: Downloading video feed
WARNING: [TikTok] 7096644298396978475: Unable to find video in feed; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U; trying with webpage
[TikTok] 7096644298396978475: Downloading webpage
ERROR: [TikTok] 7096644298396978475: Unable to download webpage: HTTP Error 404: Not Found (caused by <HTTPError 404: 'Not Found'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/Users/julian/yt-dlp/yt-dlp/yt_dlp/extractor/common.py", line 672, in extract
    ie_result = self._real_extract(url)
  File "/Users/julian/yt-dlp/yt-dlp/yt_dlp/extractor/tiktok.py", line 541, in _real_extract
    webpage = self._download_webpage(url, video_id, headers={'User-Agent': 'User-Agent:Mozilla/5.0'})
  File "/Users/julian/yt-dlp/yt-dlp/yt_dlp/extractor/common.py", line 1081, in _download_webpage
    return self.__download_webpage(url_or_request, video_id, note, errnote, None, fatal, *args, **kwargs)
  File "/Users/julian/yt-dlp/yt-dlp/yt_dlp/extractor/common.py", line 1032, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "/Users/julian/yt-dlp/yt-dlp/yt_dlp/extractor/common.py", line 866, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "/Users/julian/yt-dlp/yt-dlp/yt_dlp/extractor/common.py", line 823, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/Users/julian/yt-dlp/yt-dlp/yt_dlp/extractor/common.py", line 805, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "/Users/julian/yt-dlp/yt-dlp/yt_dlp/YoutubeDL.py", line 3680, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 634, in http_response
    response = self.parent.error(
  File "/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
```

**Comment by pukkandan** [comment]
cc @bashonly

**Comment by julian45** [comment]
To clarify the initial report, and also to help diagnosis/remediation: [douyin.wtf](https://douyin.wtf/), which is a public instance of [this repo](https://github.com/Evil0ctal/Douyin_TikTok_Download_API), is able to find the video with sound on TikTok's CDN. If you feed it the TikTok URL I tested for this report, although the first download links on the resulting page don't appear to lead to the correct video, the bottom link in the results takes you to the [raw API response](https://api.douyin.wtf/api?url=https://www.tiktok.com/@espnatbat/video/7096644298396978475), whose metadata includes URLs for the correct (I think) unwatermarked video (`nwm_video_url`) and watermarked video (`wm_video_url`), as well as a bunch of other metadata.

tl;dr: even if the sound is "removed" and the link 404s in a browser, the video and audio still exist on TikTok's CDN and are somehow programmatically findable.

**Comment by bashonly** [comment]
> To clarify the initial report, and also to help diagnosis/remediation: [douyin.wtf](https://douyin.wtf/), which is a public instance of [this repo](https://github.com/Evil0ctal/Douyin_TikTok_Download_API), is able to find the video with sound on TikTok's CDN. If you feed it the TikTok URL I tested for this report, although the first download links on the resulting page don't appear to lead to the correct video, the bottom link in the results takes you to the [raw API response](https://api.douyin.wtf/api?url=https://www.tiktok.com/@espnatbat/video/7096644298396978475), whose metadata includes URLs for the correct (I think) unwatermarked video (`nwm_video_url`) and watermarked video (`wm_video_url`), as well as a bunch of other metadata.
> 
> tl;dr: even if the sound is "removed" and the link 404s in a browser, the video and audio still exist on TikTok's CDN and are somehow programmatically findable.

this repo uses basically the same feed workaround as yt-dlp's extractor, except it does not check if the feed result matches the requested video ID. the `douyin.wtf` API response is pointing to the wrong video. don't think there is anything new to be learned from this one, unfortunately.

appreciate the tip though, and the proper issue

**Comment by EveryDirection** [comment]
Just wanted to add independent confirmation of this issue of yt-dlp being unable to download videos with sound removed. I encounter this problem fairly regularly and hope for a fix.

**Comment by pointydev** [comment]
Hey there, just ran into this issue, and I thought I should report that I found it appears audio can be obtained when viewing the embed URL:
- Audio removed: https://www.tiktok.com/@kearamr/video/6869073308583791878
- Audio available: https://www.tiktok.com/embed/v2/6869073308583791878
